runtime/tests/checker/reference_test.go

```diff
@@ -830,17 +827,15 @@ func TestCheckInvalidReferenceResourceLoss(t *testing.T) {
       fun test() {
           let rs <- [<-create R()]
           let ref = &rs as &[R]
-          ref[0]
+          ref[0]  // This result in a reference, so no resource loss
```

Comment:
Do we still have a case for the non-reference / owned case?

Reply:
Deja-vu: https://github.com/onflow/cadence/pull/2573#discussion_r1234448193 😄